packages/angular_devkit/build_angular/src/builders/jest/index.ts

```diff
@@ -0,0 +1,147 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import { BuilderContext, BuilderOutput, createBuilder } from '@angular-devkit/architect';
+import { execFile as execFileCb } from 'child_process';
+import * as path from 'path';
+import { promisify } from 'util';
+import { buildEsbuildBrowserInternal } from '../browser-esbuild';
+import { BrowserEsbuildOptions } from '../browser-esbuild/options';
+import { OutputHashing } from '../browser-esbuild/schema';
+import { normalizeOptions } from './options';
+import { Schema as JestBuilderSchema } from './schema';
+import { findTestFiles } from './test-files';
+
+const execFile = promisify(execFileCb);
+
+/** Main execution function for the Jest builder. */
+export default createBuilder(
+  async (schema: JestBuilderSchema, context: BuilderContext): Promise<BuilderOutput> => {
+    context.logger.warn(
+      'NOTE: The Jest builder is currently EXPERIMENTAL and not ready for production use.',
+    );
+
+    const options = normalizeOptions(schema);
+    const testOut = 'dist/test-out'; // TODO(dgp1130): Hide in temp directory.
+
+    // Verify Jest installation and get the path to it's binary.
+    // We need to `node_modules/.bin/jest`, but there is no means to resolve that directly. Fortunately Jest's `package.json` exports the
+    // same file at `bin/jest`, so we can just resolve that instead.
+    const jest = require.resolve('jest/bin/jest');
+    if (!jest) {
+      return {
+        success: false,
+        error:
+          'Jest is not installed, most likely you need to run `npm install jest --save-dev` in your project.',
+      };
+    }
+
+    // Verify that JSDom is installed in the project.
+    const environment = require.resolve('jest-environment-jsdom');
+    if (!environment) {
+      return {
+        success: false,
+        error:
+          '`jest-environment-jsdom` is not installed. Install it with `npm install jest-environment-jsdom --save-dev`.',
+      };
+    }
+
+    // Build all the test files.
+    const testFiles = await findTestFiles(options, context.workspaceRoot);
+    const jestGlobal = path.join(__dirname, 'jest-global.mjs');
+    const initTestBed = path.join(__dirname, 'init-test-bed.mjs');
+    const buildResult = await build(context, {
+      // Build all the test files and also the `jest-global` and `init-test-bed` scripts.
+      entryPoints: new Set(Array.from(testFiles).concat([jestGlobal, initTestBed])),
+      tsConfig: options.tsConfig,
+      polyfills: options.polyfills ?? ['zone.js', 'zone.js/testing'],
```

Comment:
This similar to the other comment about "tsConfig". We should have consistence between builders. I am okay with adding a default as long as we know that other builders will follow suit or we are going to delete the default from this builder. Also, in this case the default is in code, unlike the one for `tsConfig`, this is not ideal in documentation/help pages it will show that by default no polyfills are included which is not the case.

Therefore, if we are adding a default this should be in the schema.json and not in code.

Reply:
After some discussion we decided to update this to align with other builders, so `polyfills` defaults to `[]` (as specified by the schema, so docs should be accurate). Users need to add `['zone.js', 'zone.js/testing']` if they are using Zones.

We should probably have a separate conversation about whether this should be the default, but decision should probably be made for _all_ official builders, not just Jest.